src/transformers/quantizers/quantizer_hqq.py

```diff
@@ -99,6 +99,52 @@ def update_missing_keys(
         else:
             return missing_keys
 
+    # Adds missing keys for HQQLinear modules that are loaded but the model with initialized with torch.nn.Linear
+    def update_expected_keys(self, model: "PreTrainedModel", keys: List[str], loaded_keys: List[str]) -> List[str]:
+        # Collects all quantizable (linear) layers
+        def _find_hqq_quantizable_layers(model, layers):
+            for name, module in model.named_children():
+                if isinstance(module, (torch.nn.Linear)):
+                    layers.add(module.name)
+                _find_hqq_quantizable_layers(module, layers)
+
+        new_keys = set(keys)
+        if is_hqq_available():
+            from hqq.core.quantize import HQQLinear
```

Comment:
add a `if self.pre_quantized` condition as we shouldn't modify the keys when the model is not quantized yet.

Reply:
Sure, works with the check! https://github.com/huggingface/transformers/pull/33141/commits/9053ad5f4c4bea7e396f12896d072c0999dda74c